src/BinOp.chpl

```diff
@@ -1044,4 +1045,585 @@ module BinOp
+    }
+    else {
+      if has_max_bits {
+        // max_size should always be non-zero since we start at 1 and left shift
+        tmp.mod(tmp, max_size);
+      }
+      return (tmp, max_bits);
+    }
+  }
+
+  proc doBigIntBinOpvvBoolReturn(l, r, op: string) throws {
+    select op {
+      when "<" {
+        return l.a < r.a;
+      }
+      when ">" {
+        return l.a > r.a;
+      }
+      when "<=" {
+        return l.a <= r.a;
+      }
+      when ">=" {
+        return l.a >= r.a;
+      }
+      when "==" {
+        return l.a == r.a;
+      }
+      when "!=" {
+        return l.a != r.a;
+      }
+      otherwise {
+        // we should never reach this since we only enter this proc
+        // if boolOps.contains(op)
+        throw new Error("Unsupported operation: " + l.etype:string +" "+ op +" "+ r.etype:string);
+      }
+    }
+  }
+
+  proc doBigIntBinOpvs(l, val, op: string) throws {
+    var max_bits = l.max_bits;
+    var max_size = 1:bigint;
+    var has_max_bits = max_bits != -1;
+    if has_max_bits {
+      max_size <<= max_bits;
+    }
+    ref la = l.a;
+    var tmp = makeDistArray(la.size, bigint);
+    // i think tmp should be initialized to 0 but just to be sure
+    tmp = 0:bigint;
+    // these cases are not mutually exclusive,
+    // so we have a flag to track if tmp is ever populated
+    var visted = false;
+
+    // had to create bigint specific BinOp procs which return
+    // the distributed array because we need it at SymEntry creation time
+    if l.etype == bigint && val.type == bigint {
+      // first we try the ops that only work with
+      // both being bigint
+      select op {
+        when "&" {
+          tmp = la & val;
+          visted = true;
+        }
+        when "|" {
+          tmp = la | val;
+          visted = true;
+        }
+        when "^" {
+          tmp = la ^ val;
+          visted = true;
+        }
+        when "/" {
+          tmp = la / val;
+          visted = true;
+        }
+      }
+    }
+    if l.etype == bigint && (val.type == bigint || val.type == int || val.type == uint) {
+      // then we try the ops that only work with a
+      // left hand side of bigint
+      if val.type != bigint {
+        // can't shift a bigint by a bigint
+        select op {
+          when "<<" {
+            tmp = la << val;
+            visted = true;
+          }
+          when ">>" {
+            // workaround for right shift until chapel issue #21206
+            // makes it into a release, eventually we can just do
+            // tmp = la >> ra;
+            var divideBy = makeDistArray(la.size, bigint);
+            divideBy = 1:bigint;
+            divideBy <<= val;
+            tmp = la / divideBy;
+            visted = true;
+          }
+          when "<<<" {
+            if !has_max_bits {
+              throw new Error("Must set max_bits to rotl");
+            }
+            // return (la << val) | (la >> (max_bits - val));
```

Comment:
Again, I marked this in other functions, but is this waiting on the same fix provided by Chapel #21206?